dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceCollectionExtensions.cs

```diff
@@ -213,14 +214,125 @@ public static class OpenAIServiceCollectionExtensions
                 serviceProvider.GetService<ILoggerFactory>()));
     }
 
+    /// <summary>
+    /// Adds an Azure OpenAI text generation service with the specified configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="IKernelBuilder"/> instance to augment.</param>
+    /// <param name="config">Required configuration for Azure OpenAI.</param>
+    /// <param name="serviceId">A local identifier for the given AI service.</param>
+    /// <param name="httpClient">The HttpClient to use with this service.</param>
+    /// <returns>The same instance as <paramref name="builder"/>.</returns>
+    public static IKernelBuilder AddAzureOpenAITextGeneration(
+        this IKernelBuilder builder,
+        AzureOpenAIConfig config,
+        string? serviceId = null,
+        HttpClient? httpClient = null)
+    {
+        Verify.NotNull(builder);
+        Verify.NotNull(config);
+
+        config.Validate();
+
+        switch (config.Auth)
```

Comment:
Let's avoid using enums for this configuration. I think we can create hierarchy of configuration classes, where base class will contain common properties like `deploymentName` etc, and different derived types will contain configuration-specific properties like `TokenCredential` or `string ApiKey`. I think the main goal would be to create the same set of overloads we have today, but with specific config models. This approach will also decrease the probability of incorrect configuration when `AuthType` is unknown. With specific classes we won't need to throw an exception.

Otherwise, if enum value is `TokenCredential`, we will have to verify that the property `TokenCredential` was initialized and so on. The more parameters will be added in the future, the harder it will be to maintain the validation. But having specific config models should simplify the usage and the maintenance.

Reply:
Please see my [comment](https://github.com/microsoft/semantic-kernel/pull/6283#issuecomment-2117053217) below.

Reply:
Adjusted as requested. Please take a look.